dll/win32/comctl32/edit.c

```diff
@@ -2592,7 +2592,14 @@ static void EDIT_EM_ReplaceSel(EDITSTATE *es, BOOL can_undo, const WCHAR *lpsz_r
 		/* remove chars that don't fit */
 		if (honor_limit && !(es->style & ES_AUTOHSCROLL) && (es->text_width > fw)) {
 			while ((es->text_width > fw) && s + strl >= s) {
-				lstrcpyW(es->text + s + strl - 1, es->text + s + strl);
+				int len = wcslen(es->text);
+                for (int i = s + strl - 1; i < len; i++) 
```

Comment:
So that we can understand the problem: Is it because `s + strl == 0` that there is the problem? (Maybe because both `s` and `strl` are zero?)

If so, maybe the simpler fix could be something in the condition of the while-loop just above? For example:
`s + strl > s` ,  or `s + strl >= s && s > 0` (whichever is correct, you would have to test with a debugger).

Reply:
The problem is, that (at least in the case of CORE-19743), `s + strl == 0` is true before the loop even starts. As such, the loop would not be executed. I not sure why exactly this code is here in the first place - does Windows truncate text in singleline non-scroll edit boxes (and even more bizarre is the fact that this code truncates from the start)? This is even mentioned in the JIRA issue as an unexpected behaviour. Not knowing what the reason for this code being here, I decided to follow its logic as closely as possible, only changing it, so it doesn't corrupt the heap.